### PR TITLE
Add ssp tx fifo flush

### PR DESCRIPTION
**Pull request by plbossart**
Trying to see if this addresses @jwrdegoede's feedback of 'crackle noise' on startup.

### CODE REVIEW

**Comment by lgirdwood** [comment]
@plbossart the alsa-bat is passing. But there is no WAV to download.
```
Test Case URL: https://github.com/thesofproject/sof-test/tree/master/test-case/check-alsabat.sh
cmd: TPLG=sof-cht-max98090.tplg ~/sof-test/test-case/check-alsabat.sh -p hw:max98090,0 -c hw:Device,0
====================
framework load: ssh  ubuntu@jf-bsw-cyn-max98090-8.jf.intel.com  'nohup bash -c "TPLG=sof-cht-max98090.tplg ~/sof-test/test-case/check-alsabat.sh -p hw:max98090,0 -c hw:Device,0 2>&1 | nc -N 10.239.156.45 44977" >/dev/null &'
====================
framework run: 'netcat -l -k 44977'
====================
2020-12-02 19:48:26 UTC [REMOTE_INFO] Starting /usr/bin/sof-logger -t -l /etc/sof/sof-cht.ldc -o /home/ubuntu/sof-test/logs/check-alsabat/2020-12-02-19:48:26-5199/slogger.txt
2020-12-02 19:48:26 UTC [REMOTE_INFO] check the PCMs before alsabat test
2020-12-02 19:48:28 UTC [REMOTE_COMMAND] alsabat -Pplughw:max98090,0 --standalone -n 240000 -F 997
2020-12-02 19:48:29 UTC [REMOTE_COMMAND] alsabat -Cplughw:Device,0 -F 997
alsa-utils version 1.2.2

Entering capture thread (ALSA).
Get period size: 2756  buffer size: 22050
Recording ...
Capture completed.

BAT analysis: signal has 65536 frames at 44100 Hz, 1 channels, 2 bytes per sample.

Channel 1 - Checking for target frequency 997.00 Hz
Amplitude: 8324.5; Percentage: [25]
Detected peak at 996.58 Hz of 35.43 dB
 Total 41.3 dB from 985.14 to 1008.70 Hz
 PASS: Peak detected at target frequency
Detected at least 1 signal(s) in total

Return value is 0
alsa-utils version 1.2.2

Entering playback thread (ALSA).
Get period size: 1881  buffer size: 15052
Playing generated audio sine wave
Playback completed.

Return value is 0
2020-12-02 19:48:34 UTC [REMOTE_INFO] pkill -TERM sof-logger
Terminated
2020-12-02 19:48:35 UTC [REMOTE_INFO] nlines=2282 /home/ubuntu/sof-test/logs/check-alsabat/2020-12-02-19:48:26-5199/slogger.txt
2020-12-02 19:48:36 UTC [REMOTE_INFO] Test Result: PASS!
```

File: src/drivers/intel/baytrail/ssp.c [hunk 1]
```diff
@@ -525,10 +547,13 @@ static void ssp_start(struct dai *dai, int direction)
 	dai_info(dai, "ssp_start()");
 
 	/* enable DMA */
-	if (direction == DAI_DIR_PLAYBACK)
+	if (direction == DAI_DIR_PLAYBACK) {
 		ssp_update_bits(dai, SSCR1, SSCR1_TSRE, SSCR1_TSRE);
-	else
+		ssp_update_bits(dai, SSTSA, SSTSA_TXEN, SSTSA_TXEN);
```
Comment (keyonjie) [inline src/drivers/intel/baytrail/ssp.c:552]: iiuc, we don't have TXEN/RXEN bit on BYT SSP, which means we can't stop TX/RX separately there, if you do a while() check to the TFL/RFL in the FIFO flush, you will find this sad result.

Comment (plbossart) [reply src/drivers/intel/baytrail/ssp.c:552]: sorry @keyonjie not understanding your feedback. 
There's clearly separate registers for SSTSA_TXEN and SSRSA_RXEN.

What are you referring to?

Comment (keyonjie) [reply src/drivers/intel/baytrail/ssp.c:552]: @plbossart I guess you are referring to the wrong register map, for BYT, we don't have SSTSA_TXEN and SSRSA_TXEN bits, though we have TTSA abd RTSA there.

Comment (plbossart) [reply src/drivers/intel/baytrail/ssp.c:552]: @keyonjie I checked the hardware documentation.

Comment (plbossart) [reply src/drivers/intel/baytrail/ssp.c:552]: Actually now I am confused, I only see RTSA and TTSA but I don't see where we set them. Humm.

**Comment by paulstelian97** [comment]
Is this still needed?

**Pull request closed without merging**